aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -255,6 +255,7 @@ impl<'r> WriteOpConverter<'r> {
             self.convert(state_value_metadata, metadata_op, false)?,
             inner_ops,
             post_group_size.get(),
+            pre_group_size.get(),
```

Comment:
@msmouse if you don't mind, and are making changes to the PR anyway (otherwise i'll do it later), could you update the comment above, just noticed it's outdated (we don't encode the size, also an important point is to explain why we even need an Op now).
i.e. could you please replace 243-246 with 
"Create an op to encode the proper kind for resource group operation"
(@igor-aptos I think now this is the main reason?)
Interestingly, we may be able to get rid of the ugly metadata_op, since we have both pre_group_size and post_group_size, thanks @msmouse! (but not in this PR obviously :)

Otherwise, pre_group_size should be properly validated, we rely on it for resource group correctness, in fact the kind, i.e. Creation/Modification/Deletion, and it passed replay.
proptests for resource groups do already check the resource_group_size() call (re: @igor-aptos)